src/utils.ts

```diff
@@ -51,3 +62,78 @@ const weekdayNamesValues = Object.values(WEEKDAY_NAMES);
 export function isValidWeekdayName(weekday: unknown): weekday is WeekdayName {
   return weekdayNamesValues.includes(weekday as WeekdayName);
 }
+
+const monthNamesValues = Object.values(MONTH_NAMES);
+const monthNamesEntries = Object.entries(MONTH_NAMES);
+
+export function isValidMonthName(month: unknown): month is MonthName {
+  return monthNamesValues.includes(month as MonthName);
+}
+
+export function getMonthIndex(monthName: MonthName): number {
+  const monthNamesEntry = monthNamesEntries.find(([, value]) => value === monthName);
+
+  if (!monthNamesEntry) {
+    throw new Error(`Invalid month name: ${monthName}`);
+  }
+
+  return Number(monthNamesEntry[0]);
+}
+
+// Maximum days per month (for validation, not accounting for leap years)
+const MAX_DAYS_PER_MONTH: Record<MonthName, number> = {
+  Jan: 31,
+  Feb: 29, // Allow 29 for leap years
+  Mar: 31,
+  Apr: 30,
+  May: 31,
+  Jun: 30,
+  Jul: 31,
+  Aug: 31,
+  Sep: 30,
+  Oct: 31,
+  Nov: 30,
+  Dec: 31,
+};
+
+export function isValidDayOfMonth(day: number, month: MonthName): boolean {
+  if (!Number.isInteger(day) || day < 1) {
+    return false;
+  }
+
+  const maxDays = MAX_DAYS_PER_MONTH[month];
+  return day <= maxDays;
+}
+
+export function isRecurringOpeningHours(
+  openingHours: OpeningHours,
+): openingHours is RecurringOpeningHours {
+  return 'from' in openingHours && 'to' in openingHours;
+}
+
+export function isAbsoluteOpeningHours(
+  openingHours: OpeningHours,
+): openingHours is AbsoluteOpeningHours {
+  return 'dates' in openingHours;
+}
+
+export function matchesAbsoluteDate(date: Date, absoluteDateStr: string): boolean {
+  const match = absoluteDateStr.match(/^([A-Z][a-z]{2})\s+(\d+)$/);
+  if (!match?.[1] || !match[2]) {
+    return false;
+  }
+
+  const monthName = match[1] as MonthName;
+  const day = Number.parseInt(match[2], 10);
+  const month = getMonthIndex(monthName);
+
+  return date.getMonth() === month && date.getDate() === day;
+}
+
+export function matchesAnyAbsoluteDate(date: Date, absoluteDates: string[]): boolean {
+  return absoluteDates.some((absoluteDate) => matchesAbsoluteDate(date, absoluteDate));
+}
+
+export function getMatchingAbsoluteDate(date: Date, absoluteDates: string[]): string | undefined {
+  return absoluteDates.find((absoluteDateStr) => matchesAbsoluteDate(date, absoluteDateStr));
```

Comment:
Rename absoluteDateStr to just absoluteDate to keep it in line with other places